src/v/redpanda/admin_server.cc

```diff
@@ -2812,6 +2819,64 @@ admin_server::mark_transaction_expired_handler(
       });
 }
 
+static int expect_query_int(
+  const std::unique_ptr<ss::http::request>& req, const ss::sstring& name) {
+    auto val = req->get_query_param(name);
+    if (val.empty()) {
+        throw ss::httpd::bad_param_exception(
+          fmt::format("{}:int must be provided in query_params", name));
+    }
+    try {
+        return static_cast<int>(std::stoi(val));
+    } catch (...) {
+        throw ss::httpd::bad_param_exception(
+          fmt::format("{} must be int; got: {}", name, val));
+    }
+}
+
+ss::future<ss::json::json_return_type>
+admin_server::mark_transactions_draining_handler(
+  std::unique_ptr<ss::http::request> req) {
+    if (!config::shard_local_cfg().enable_transactions) {
+        throw ss::httpd::bad_request_exception("Transaction are disabled");
+    }
+
+    int tx_tm_partition = expect_query_int(req, "tm_partition");
+
+    auto content = co_await ss::util::read_entire_stream_contiguous(
+      *req->content_stream);
+    vlog(
+      logger.info,
+      "mark_transactions_draining_handler request with data: {}",
+      content);
+
+    cluster::draining_txs req_data;
+    try {
+        req_data = pandaproxy::json::rjson_parse(
+          content.c_str(), pandaproxy::json::draining_txs_parser());
+    } catch (pandaproxy::json::parse_error e) {
+        throw ss::httpd::bad_param_exception("Invalid request data");
+    }
+
+    auto tx_ntp = model::ntp(
+      model::tx_manager_nt.ns, model::tx_manager_nt.tp, tx_tm_partition);
+
+    auto& tx_frontend = _partition_manager.local().get_tx_frontend();
+    if (!tx_frontend.local_is_initialized()) {
+        throw ss::httpd::bad_request_exception("Can not get tx_frontend");
+    }
+
+    // we need an upper boundary for happy case cluster wide replication
+    // and create_topic_timeout_ms is a good approximation
+    auto timeout = config::shard_local_cfg().create_topic_timeout_ms();
```

Comment:
this should be a separate timeout, it is confusing that the `create_topic_timeout` is used in transaction context.